### PR TITLE
Deprecate `on_before_accelerator_backend_setup` callback hook

### LINKED ISSUE #11559: [RFC] Deprecate `on_before_accelerator_backend_setup` hook
## Proposed refactor

Deprecate `on_before_accelerator_backend_setup` hook in v1.6 for removal in v1.8. Redirect users to the `setup` hook.
https://github.com/PyTorchLightning/pytorch-lightning/blob/5ad5ba54c0c477546d21daf75ac7b4748d7963a7/pytorch_lightning/callbacks/base.py#L63-L65

### Motivation

I only see two external uses of this hook and 3 internal uses in https://www.grep.app/search?q=on_before_accelerator_backend_setup 
Users can implement the `setup` hook instead.

### Additional context

This hook was originally added in #5377
Part of #7740

cc @justusschock @awaelchli @akihironitta @rohitgr7 @tchaton @carmocca @borda @ninginthecloud @daniellepintz

### PR DESCRIPTION
**Pull request by krishnakalyan3**
## What does this PR do?

Fixes #11559

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by krishnakalyan3** [comment]
@ananthsub this is part 2 of [this](https://github.com/PyTorchLightning/pytorch-lightning/pull/11568) PR. Does it look okay?

**Comment by daniellepintz** [comment]
@krishnakalyan3 currently this PR is only changing the changelog. It should be deprecating the hook, like in https://github.com/PyTorchLightning/pytorch-lightning/pull/10940

Please also update the title to be the same as #11559

File: CHANGELOG.md [hunk 1]
```diff
@@ -790,6 +790,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `PrecisionPlugin.master_params()` in favor of `PrecisionPlugin.main_params()` ([#10105](https://github.com/PyTorchLightning/pytorch-lightning/pull/10105))
 - Deprecated `lr_sch_names` from `LearningRateMonitor` ([#10066](https://github.com/PyTorchLightning/pytorch-lightning/pull/10066))
 - Deprecated `ProgressBar` callback in favor of `TQDMProgressBar` ([#10134](https://github.com/PyTorchLightning/pytorch-lightning/pull/10134))
+- Deprecated `on_before_accelerator_backend_setup`callback ([#11568](https://github.com/PyTorchLightning/pytorch-lightning/pull/11568))
```
Comment (daniellepintz) [inline CHANGELOG.md:793]: This section is for what was deprecated in v1.5. Please move this line up to ~282

**Review by ananthsub — CHANGES_REQUESTED** [review]
Requesting changes based on @daniellepintz 's feedback

**Review by tchaton — APPROVED** [review]
LGTM once comments are addressed.

**Comment by daniellepintz** [comment]
The title and description of this PR don't match the contents - you seem to be removing the `on_pretrain_routine_start` hooks

**Comment by krishnakalyan3** [comment]
@daniellepintz You are right. Now, this PR only contains the change log.

**Comment by krishnakalyan3** [comment]
Part 2/2 #11550

File: pytorch_lightning/trainer/configuration_validator.py [hunk 2]
```diff
@@ -348,3 +350,12 @@ def _check_on_configure_sharded_model(trainer: "pl.Trainer") -> None:
                 "The `on_configure_sharded_model` callback hook was deprecated in"
                 " v1.6 and will be removed in v1.8. Use `setup()` instead."
             )
+
+
+def _check_on_before_accelerator_backend_setup(trainer: "pl.Trainer", model: "pl.LightningModule") -> None:
```
Comment (ananthsub) [inline pytorch_lightning/trainer/configuration_validator.py:355]: `model` isn't used - remove it from the function signature

File: tests/deprecated_api/test_remove_1-8.py [hunk 3]
```diff
@@ -447,3 +447,25 @@ def test_v1_8_0_rank_zero_imports():
...
+        fast_dev_run=True,
+        enable_progress_bar=False,
+        logger=False,
+        default_root_dir=tmpdir,
+    )
+    with pytest.deprecated_call(
+        match="The `on_before_accelerator_backend_setup` callback hook was deprecated in v1.6 and will be removed in v1.8"
+    ):
+        trainer.fit(model)
+        trainer.validate(model)
```
Comment (ananthsub) [inline tests/deprecated_api/test_remove_1-8.py:471]: you only need to call one of these functions for the test

**Review by ananthsub — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Comment by krishnakalyan3** [comment]
Thanks @rohitgr7

File: tests/callbacks/test_finetuning_callback.py [hunk 4]
```diff
@@ -264,8 +264,8 @@ def configure_optimizers(self):
 
 
 def test_on_before_accelerator_backend_setup(tmpdir):
```
[the remark this answers is not in the document]
Comment (carmocca) [reply tests/callbacks/test_finetuning_callback.py:266]: Probably not, it's also covered by `tests/models/test_hooks.py::test_trainer_model_hook_system_fit`

**Pull request merged**